### PR TITLE
Integration test for ProxyStatus and Experimental AuthZ Check

### PR DESCRIPTION
**Pull request by esnible**
Part one of the fix for https://github.com/istio/istio/issues/23042

Once the istioctl subcommands that query Envoy are tested during integration we will remove the static unit tests for those subcommands.

cc @howardjohn

### CODE REVIEW

File: tests/integration/pilot/istioctl_test.go
Comment (howardjohn) [inline tests/integration/pilot/istioctl_test.go:329]: why do we need dummy namespace?

File: tests/integration/pilot/istioctl_test.go
Comment (howardjohn) [inline tests/integration/pilot/istioctl_test.go:369]: This doesn't really test much though.. The code could be completely wrong and not catch this.

File: tests/integration/pilot/istioctl_test.go
Comment (howardjohn) [inline tests/integration/pilot/istioctl_test.go:347]: why not have a single pod shared between the tests then just run a serious of `istioctl` commands against it?

**Review by howardjohn — COMMENTED** [review]
overall looks good

File: tests/integration/pilot/istioctl_test.go
Comment (esnible) [reply tests/integration/pilot/istioctl_test.go:347]: I'd like to do that.  Is there a way that I can have multiple TestXYX() functions run after the same pod is ready?  Or is it expected that I have a single Test() function and call out to functions for each subcommand?

Comment (howardjohn) [reply tests/integration/pilot/istioctl_test.go:349]: Here is how you could do it (pseudocode, may be slightly off)

```
func TestIstioctl() {
  framework.NewTest.Run(func() {
    ns := ...
   echoInstance := setupInstance
   ctx.NewSubtest("describe").Run(func() { ... run istioctl test ...})
   ctx.NewSubtest("authz").Run(func() { ... run istioctl test ...})
}
```

File: tests/integration/pilot/istioctl_test.go
Comment (howardjohn) [inline tests/integration/pilot/istioctl_test.go:370]: Not in scope for this PR, but this is exposing a bug. I am pretty sure its not supposed to say `noneSDS: default`

**Review by howardjohn — APPROVED** [review]
LGTM, can we merge these as a followup and probably also file a bug for authz output?

File: tests/integration/pilot/istioctl_test.go
Comment (esnible) [reply tests/integration/pilot/istioctl_test.go:349]: I believe these tests already run in parallel.   I like the look of that and will use that style in a follow-up.

Comment (howardjohn) [reply tests/integration/pilot/istioctl_test.go:349]: You can chose parallel or serial with Run or RunParallel I think. Parallel seems reasonable here

**Comment by esnible** [comment]
Can anyone from @istio/wg-user-experience-maintainers review this for me?

**Review by sdake — APPROVED** [review]
LGTM. Refinements can be done as needed. @esnible would you file a bug to track the `ads: default` problem @howardjohn found?

**Comment by esnible** [comment]
> LGTM. Refinements can be done as needed. @esnible would you file a bug to track the `ads: default` problem @howardjohn found?

Issue https://github.com/istio/istio/issues/23526 tracks that problem.

**Review by liamawhite — APPROVED** [review]

**Review by howardjohn — CHANGES_REQUESTED** [review]
looks like a bad merge

**Comment by esnible** [comment]
@googlebot I fixed it.

**Comment by esnible** [comment]
I fixed it.  Question.  I did `git rebase upstream/master`, made some changes, added, committed, pushed.
The PR report hundreds of files chosen.  You commented there was a bad merge.

To get it back to 2 files, I clicked "Edit" near the title, changed to another base, did it again, to change it back to be based on `master`.  That made the PR look normal.  What `git` command should I have used instead?

**Comment by esnible** [comment]
This needs to go on hold until https://github.com/istio/istio/pull/23724 merges.  A performance improvement in `istioctl proxy-status <pod>` is keeping this test from passing.  When that PR merges it it will be safe to retest this one.

**Comment by esnible** [comment]
Added a fix that will also resolve https://github.com/istio/istio/issues/23735

**Comment by esnible** [comment]
@howardjohn Can you re-review?

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
